### PR TITLE
tweak: DUP headsign replacement

### PR DESCRIPTION
**Pull request by cmaddox5**
**Asana task**: [Add new DUP headsign abbreviation for "Middleborough/Lakeville"](https://app.asana.com/0/1185117109217413/1203658337706587/f)

Instead of adding a new case for every headsign with a `/`, I added a `String.replace` call that looks for a `/` without spaces around it. Needed to use regex to make it happen, but this should keep us from needing a task for each new headsign.

- [ ] Tests added?

### CODE REVIEW

File: lib/screens/dup_screen_data/request.ex [hunk 1]
```diff
@@ -208,7 +207,13 @@ defmodule Screens.DupScreenData.Request do
   end
 
   defp replace_long_headsigns(%{destination: destination} = departure_map) do
-    replaced_destination = Map.get(@headsign_replacements, destination, destination)
+    replaced_destination =
+      @headsign_replacements
+      |> Map.get(destination, destination)
+      # Find any headsign with a slash that does NOT have spaces around it and add spaces
+      # i.e. String.replace("Middleborough/Lakeville", ~r/([^[:space:]])([\/])([^[:space:]])/, "\\1 / \\3") => "Middleborough / Lakeville"
+      |> String.replace(~r/([^[:space:]])([\/])([^[:space:]])/, "\\1 / \\3")
```
Comment (PaulJKim) [inline lib/screens/dup_screen_data/request.ex:215]: Don't know how often this edge case would appear or how important it is that we handle it, but if the slash has a space only on one side of it, for example `test/ string", the regex fails to catch that and add a space to the one side that doesn't have the space.

Comment (cmaddox5) [reply lib/screens/dup_screen_data/request.ex]: I'm not sure either so I went ahead and tweaked the regex to cover it. The most common case (a slash with no spaces on either side) is still covered. Now a slash only has a space on the right side, it will still add a space on both sides. This will cause an extra space on the right side, but the front end splits the string on whitespace so any extra whitespace is trimmed.

Comment (PaulJKim) [reply lib/screens/dup_screen_data/request.ex]: Should we just add a space on both sides of the slash? I think this doesn't cover the converse case where the slash only has a space on the left. Or would the frontend handle that case alright as is?

Comment (cmaddox5) [reply lib/screens/dup_screen_data/request.ex]: Yeah that's a good point. Always adding a space on each side doesn't cause extra spaces. The frontend string splitting always trims them.

Comment (jzimbel-mbta) [reply lib/screens/dup_screen_data/request.ex]: The regex could be slightly adjusted to capture just the slash and 0 or 1 space on either side of it, and then you could simply replace that single capture with `" / "`:
```ex
~r/[[:space:]]?\/[[:space:]]?/
```

```ex
iex> tests = ["a/b", "a /b", "a/ b", "a / b"]
iex> Enum.map(tests, &String.replace(&1, ~r/([[:space:]]?\/[[:space:]]?)/, " / "))
["a / b", "a / b", "a / b", "a / b"]
```

Per regex101.com:
image

Comment (cmaddox5) [reply lib/screens/dup_screen_data/request.ex]: Exactly what we need! Thanks Jon!

Comment (jzimbel-mbta) [reply lib/screens/dup_screen_data/request.ex]: You can also use `\s` instead of `[[:space:]]`! I'd never checked until now, but from a few searches it appears they represent an identical set of whitespace characters.

**Review by PaulJKim — APPROVED** [review]
LGTM!

File: lib/screens/dup_screen_data/request.ex (hunk 1 above)
Comment (cmaddox5) [reply lib/screens/dup_screen_data/request.ex]: Yep, I learned that today too. I can't decide if the character classes make things more or less readable though. The word is nice but the extra characters muddy up the regex a bit.
